test/sln.test.js

```diff
@@ -188,6 +187,17 @@ describe('sln', () => {
         const parseOptions = { deepParse: true };
         assert.throws(() => sln.parseSolutionSync(text, parseOptions));
       });
+
+      it('should merge package data on deep parse', () => {
```

Comment:
Thanks for adding this test too, they're really important for parsing. Based on my feedback above, I'm not sure if this test is still needed in it's current form, though you might be able to repurpose it for the "prioritize csproj over packages.config" suggestion